### DISCUSSION TITLE
Why did Uber make H3 available to everyone for free?

### DISCUSSION THREAD

**Discussion by jonmdev** [uber/h3#940] [General]
I learned of H3 a few years ago and there is something that has puzzled me ever since. Why did Uber make it freely available to everyone?

I am not asking for any "formal company statement." I am just someone who has used it. And I am curious, philosophically what might drive such a decision. Even speculatively.

Uber is obviously a multibillion dollar company. H3 is a pretty good innovation they must have spent a fair amount of money developing and designing. 

So why share it with everyone for free? It is not like this is a system that needed bug testing and the community thus performed this for them, helping them. And it is not a "protocol" that if it gains wide adoption will help Uber by making it a "standard" (like a file transfer or video codec or social media protocol might).

So what was the point of giving it away to everyone for free?

I have a hard time believing any large company does anything solely for the charity of it. 

Similarly, I sometimes see companies have their engineers write articles on Medium or elsewhere explaining unique technology choices they made or how they solved certain unique problems, which I also find strange. I would expect them to protect their IP.

Is this type of behavior a marketing gimmick to attract more good engineers? Ie. If smart engineers look at H3 or articles like I describe, they might think "Wow, ____ sounds like an interesting place to work - I should apply for a job!" and it drives hiring talent? That is one guess.

The other guess I have (particular to H3 specifically) is that perhaps the engineers or mathematicians who designed the protocols were university professors or others who had as preconditions for their work on it that it would be under public license, requiring the sharing of the result from Uber. Maybe perhaps.

I know this is not an actual "H3 Bug Issue" but I can't think of any other place to ask and it has been puzzling me for years now. StackOverflow won't allow speculative questions. I wonder if anyone here could even guess what the rationale for such "giving away valuable IP" behavior, in general, or specific to H3, might be.

Thanks for any thoughts.

**Comment by nrabinowitz** [comment]
Converted this to a discussion, as it's not a software issue. You won't get a formal company statement, since most of the H3 maintainers no longer work for Uber.

I won't give a comprehensive answer, but I can point to a few reasons:
- H3 is not a standard per se, but like a standard or protocol it is more valuable the more systems that support it. Without making H3 open source, we would likely not have H3 support in PostgreSQL, Snowflake, Rust, or various other contexts, all of which are potentially valuable to all H3 users, including Uber. To a lesser but non-negligible extent, data providers using H3 (e.g. public population data from Kontur) add to an ecosystem that all H3 users benefit from.
- We have definitely seen bug reports, bug fixes, performance improvements, and other community contributions that all H3 users, including Uber as a major consumer of the library, get for free. Even well-funded projects can benefit from the input of a broad user community.
- Uber definitely gets a brand benefit, especially in the engineering community, from discussing and releasing high-quality software. This helps to establish companies like Uber as centers of excellence, which then helps with recruiting more and better engineers. It's also frequently useful for companies to be able to hire people who are already familiar with parts of their tech stack.
- Perhaps most importantly from a practical standpoint, software projects at large companies are often pushed to open source release by individual contributors at the company who want to see their work reach a larger audience, want to be associated with a high-profile public project, and/or want to continue using and contributing to these projects after they leave the company. Companies (or individual managers) may want to support these engineers to keep them happy, or may simply not see open source as a problem. There could be some tension here with companies safeguarding their IP, which is why most large companies open-source projects that are not seen as providing a key competitive advantage, where the benefits of open source outweigh the business risk.

In the case of H3, I think the cause was primarily due to a group of engineers who worked on the project and wanted to see it become open source. And Uber has arguably benefited a great deal from this, not just from the brand halo but also from the continued maintenance of the project - in total, likely thousands of hours of dedicated engineering time from people who no longer work for the company.

**Reply by jonmdev** [reply]
Wow, thanks. That answers it all well enough for me. I appreciate you sharing your insight.
